Continuing glTF/PBR Implementation

m-agour opened the pull request:
This PR is based on the previous work in PR #715 . Trying to improve the PBR-based glTF models rendering in FURY.

It's been some time since my last contribution, and I'm excited to jump back in. 
Currently, the PBR materials are working but without Environmental lighting and reflection as in the screenshot below:

![image](https://github.com/fury-gl/fury/assets/63170874/35748934-1c6d-4b46-85a3-0ed677fcb043)

I propose merging this to establish a solid ground from which we move on with more advanced lighting for the PBR glTF models.

xtanion committed:
standard and advanced material implementation

xtanion committed:
resolving conflicts

pep8speaks commented:
Hello @m-agour, thank you for submitting the Pull Request!

* In the file [`docs/examples/viz_gltf_pbr.py`](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/docs/examples/viz_gltf_pbr.py):

> [Line 5:80](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/docs/examples/viz_gltf_pbr.py#L5): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (82 > 79 characters)

* In the file [`fury/gltf.py`](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py):

> [Line 10:1](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L10): [F401](https://duckduckgo.com/?q=pep8%20F401) 'fury.material' imported but unused
> [Line 16:1](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L16): [F401](https://duckduckgo.com/?q=pep8%20F401) 'fury.lib.PolyDataTangents' imported but unused
> [Line 16:80](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L16): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (91 > 79 characters)
> [Line 282:80](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L282): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 404:9](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L404): [F841](https://duckduckgo.com/?q=pep8%20F841) local variable 'e' is assigned to but never used
> [Line 405:19](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L405): [F541](https://duckduckgo.com/?q=pep8%20F541) f-string is missing placeholders
> [Line 437:80](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L437): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 455:80](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L455): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (80 > 79 characters)
> [Line 461:80](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L461): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (84 > 79 characters)
> [Line 537:5](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L537): [E303](https://duckduckgo.com/?q=pep8%20E303) too many blank lines (2)
> [Line 559:80](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L559): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (81 > 79 characters)
> [Line 572:24](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/gltf.py#L572): [W291](https://duckduckgo.com/?q=pep8%20W291) trailing whitespace

* In the file [`fury/lib.py`](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/lib.py):

> [Line 29:1](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/lib.py#L29): [F401](https://duckduckgo.com/?q=pep8%20F401) 'vtkmodules.util.numpy_support' imported but unused
> [Line 29:1](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/lib.py#L29): [F401](https://duckduckgo.com/?q=pep8%20F401) 'vtkmodules.util.colors' imported but unused
> [Line 30:1](https://github.com/fury-gl/fury/blob/4bec13c4c978fc2d3344d672dbffa89ba1f648ec/fury/lib.py#L30): [F401](https://duckduckgo.com/?q=pep8%20F401) 'vtkmodules.util.misc.calldata_type' imported but unused

To test for issues locally, `pip install flake8` and then run `flake8 fury`.

m-agour commented:
So, Tests are failing even though objects are being rendered, I suspect its because I modified the lightening so when it tests for a specific color it might not be there anymore. i.e. this duck produced by the test code triggers an error even though it seems fine.
![image](https://github.com/fury-gl/fury/assets/63170874/03da1009-bc44-42c9-8a81-724e9fd2ec37)

m-agour committed:
Fixed TexCoords being incorrect

m-agour committed:
Added some lighting Properties.

Also organized materials a little.

m-agour committed:
Some pep fixes

m-agour committed:
Some other pep issues

skoudoro commented:
Thank you for this @m-agour.

I will be able to look into this only after December 27. I will keep you updated. 

Thanks again!

m-agour commented:
Hello @skoudoro, please take your time. I'll keep enhancing this  it and troubleshooting the tests meanwhile. 
Happy holidays!

m-agour committed:
Fixed scene exporting not including textures

m-agour committed:
fix: included base texture

m-agour committed:
fix: updated colors in tests to match new light prefs

m-agour committed:
Merge remote-tracking branch 'upstream/master' into Fixing-PBR-for-glTF-loader

m-agour committed:
fix: interactive to False

m-agour committed:
fix: test updated to check for color match between exported and imported scene

Revert "fix: test updated to check for color match between exported and imported scene"

This reverts commit 67096f5500c6d3b43a1dcd828bd869db9b004422.

fix: test updated to check for color match between exported and imported scenes

m-agour committed:
Fix: PEP-8 issues resolved

m-agour committed:
Update tests to compare common colors

skoudoro commented:
Hi @m-agour,

Can you fix the 2 CI's "Codespell" and "Code format" before starting the review. Maybe a rebase of this PR will be enought to fix it, I did not check. Thank you in advance

m-agour commented:
Hi @skoudoro , I will rebase today and ping you after all tests pass

m-agour committed:
Merge remote-tracking branch 'upstream/master' into Fixing-PBR-for-glTF-loader

m-agour committed:
fix: codespell mistakes corrected

m-agour committed:
fix: Styling issues

m-agour committed:
fix: styling issues

m-agour committed:
Merge branch 'Fixing-PBR-for-glTF-loader' of https://github.com/m-agour/fury into Fixing-PBR-for-glTF-loader
